dspace-api/src/main/java/org/dspace/app/bulkedit/MetadataImport.java

```diff
@@ -1537,4 +1586,172 @@ public static void main(String[] argv) {
             System.exit(1);
         }
     }
+
+    /**
+     * Gets a copy of the given csv line with all entity references resolved to UUID strings.
+     *
+     * @param line the csv line to process.
+     * @return a copy, with all references resolved.
+     * @throws MetadataImportException if there is an error resolving any entity reference.
+     */
+    public DSpaceCSVLine resolveEntityRefs(DSpaceCSVLine line) throws MetadataImportException {
+        DSpaceCSVLine newLine = new DSpaceCSVLine(line.getID());
+        for (String key : line.keys()) {
+            if (key.split("\\.")[0].equalsIgnoreCase("relation")) {
+                if (line.get(key).size() > 0) {
+                    for (String val : line.get(key)) {
+                        String uuid = resolveEntityRef(c, val).toString();
+                        newLine.add(key, uuid);
+                    }
+                }
+            } else {
+                if (line.get(key).size() > 1) {
+                    for (String value : line.get(key)) {
+                        newLine.add(key, value);
+                    }
+                } else {
+                    newLine.add(key, line.get(key).get(0));
```

Comment:
I'm getting this error:
```
java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
        at java.util.ArrayList.rangeCheck(ArrayList.java:657)
        at java.util.ArrayList.get(ArrayList.java:433)
        at org.dspace.app.bulkedit.MetadataImport.resolveEntityRefs(MetadataImport.java:1613)
        at org.dspace.app.bulkedit.MetadataImport.runImport(MetadataImport.java:190)
        at org.dspace.app.bulkedit.MetadataImport.main(MetadataImport.java:1526)
```

When doing some debugging I found that:
`line.get(key)` has `[]` and `line.get(key).size()` is `0`
the key is `orgunit.identifier.vat`

and the line:
`{collection=[10400.1/4], dc.date.accessioned=[2019-04-12T13:35:18Z], dc.date.available=[2019-04-12T13:35:18Z], dc.description.provenance[en]=[Made available in DSpace on 2019-04-12T13:35:18Z (GMT). No. of bitstreams: 0], dc.identifier.uri=[http://dev5.rcaap.pt/handle/10400.1/23], orgunit.identifier.city=[Portimão], orgunit.identifier.country=[PT], orgunit.identifier.id=[403925156], orgunit.identifier.name=[Universidade do Algarve - Campus de Portimão], orgunit.identifier.ringgold=[112063], orgunit.identifier.state=[Faro], orgunit.identifier.vat=[], orgunit.identifier.zipcode=[8501-859], relation.isParentOrgUnit=[rowName:76c5cfd9-16ea-4f14-9c9c-b208da26fb29], relationship.type=[OrgUnit], rowName=[b5f709bd-ba08-47e7-9db6-b8e271ad67e0]}`

Reply:
@paulo-graca good catch -- we'll fix the issue and add a test case for a csv that contains empty/blank values

Reply:
While testing the new codebase, the error still occurred.
I think the check should be `if (!line.get(key).isEmpty()) {`

Reply:
Thanks, this works correctly now